posts/2024-05-07-24.0.0.5-beta.adoc

```diff
@@ -0,0 +1,560 @@
+
+To use this feature, add the following code to your `server.xml` file:
+
+[source,xml]
+----
+<featureManager>
+   <feature>servlet-6.1</feature>
+</featureManager>
+----
+
+For more information, see the https://jakarta.ee/specifications/servlet/6.1/[Jakarta Servlet 6.1 specification].
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/28109
+// Contact/Reviewer: jakub-pomykala,austin0,dazavala
+// // // // // // // //
+[#mq]
+== Use InstantOn with IBM® MQ in Liberty
+
+The 24.0.0.5-beta release introduces InstantOn feature support for link:{url-prefix}/docs/latest/reference/feature/messaging-3.0.html[Jakarta Messaging], link:{url-prefix}/docs/latest/reference/feature/connectors-2.1.html[Jakarta Connectors], and link:{url-prefix}/docs/latest/reference/feature/mdb-4.0.html[Jakarta Enterprise Beans Message-Driven Beans (MDB)]. InstantOn now provides blazing fast startup times for applications that use resource adapters to access external EIS resources, including applications that use the Jakarta Messaging API to access external messaging systems, like IBM® MQ. These "messaging client" applications can also manage message delivery to endpoints with message-driven bean listeners.
+
+Here's a server configuration snippet to deploy applications that use the JakartaEE-10 Messaging API and require the link:https://repo1.maven.org/maven2/com/ibm/mq/wmq.jakarta.jmsra/9.3.5.0/wmq.jakarta.jmsra-9.3.5.0.rar[IBM® MQ resource adapter] to access external messaging resources. This configuration automatically includes the `connectors-2.1` feature, which supports the resource adapter configuration element.
+
+[source,xml]
+----
+<featureManager>
+   <feature>messaging-3.1</feature>
+   <feature>mdb-4.0</feature>
+   <feature>servlet-6.0</feature>
+<featureManager/>
+
+<resourceAdapter id="mqJms" location="${server.config.dir}/wmq.jakarta.jmsra-9.3.5.0.rar"/>
+----
+
+You can use the link:https://developer.ibm.com/tutorials/mq-connect-app-queue-manager-containers/[MQ in Container image] to stand up an MQ server that provides the following resources: queue manager `QM1`, queue `DEV.QUEUE.1`, channel `DEV.APP.SVRCONN`, and listener `SYSTEM.LISTENER.TCP.1` on port `1414`.  These resources can support a simple point-to-point messaging scenario and are named within the messaging feature configuration elements shown in the following example.
+
+InstantOn can dynamically update the Messaging and Connectors feature configurations to access resources that are provided by any environment in which a server is restored from checkpoint. The following messaging configuration snippet uses the hostname and MQ listener port values (variables) defined in the restore environment. When this server restores, the connection factory and endpoint activation configurations access MQ resources by using the updated hostname and port values.

```

Comment:
I need to disambiguate this paragraph.  Please replace it with the following, which is more precise and does not muddle the usage of the HOSTNAME environment variable.  

InstantOn can dynamically update the Messaging and Connectors feature configuration elements to enable connections to external resources made available by any environment in which a server is restored from checkpoint.  In the snippet below, the Messaging configuration elements use Liberty variables named `MQ_HOSTNAME` and MQ_PORT to specify the hostname and IP port of the system on which the MQ queue manager is running.  Define these variables in the `server.env` file of a checkpointed server, and when the server is restored, the queue connection factory and message endpoint activation configurations will update to the environment-specific hostname and port values for the MQ queue manager.

Reply:
updated- with minor edits per IBM style. Lmk what you think